tuf/data/keys.go

```diff
@@ -34,38 +28,177 @@ type KeyPair struct {
 	Private []byte `json:"private"`
 }
 
-// TUFKey is the structure used for both public and private keys in TUF.
-// Normally it would make sense to use a different structures for public and
-// private keys, but that would change the key ID algorithm (since the canonical
-// JSON would be different). This structure should normally be accessed through
-// the PublicKey or PrivateKey interfaces.
-type TUFKey struct {
-	id    string
-	Type  KeyAlgorithm `json:"keytype"`
-	Value KeyPair      `json:"keyval"`
+// Keys represents a map of key ID to PublicKey object. It's necessary
+// to allow us to unmarshal into an interface via the json.Unmarshaller
+// interface
+type Keys map[string]PublicKey
+
+// UnmarshalJSON implements the json.Unmarshaller interface
+func (ks *Keys) UnmarshalJSON(data []byte) error {
+	parsed := make(map[string]tufKey)
+	err := json.Unmarshal(data, &parsed)
+	if err != nil {
+		return err
+	}
+	final := make(map[string]PublicKey)
+	for k, tk := range parsed {
+		final[k] = typedPublicKey(tk)
+	}
+	*ks = final
+	return nil
+}
+
+// KeyList represents a list of keys
+type KeyList []PublicKey
```

Comment:
This unfortunately was the result of me not thinking about uniformity enough (and not having enough of a concrete idea of this refactor) when I was implementing the root key rotations. The changelist objects for root key rotations use lists of keys rather than a map... I'm kicking myself now...